pkg/repository/v1_repository.go

```diff
@@ -88,13 +92,100 @@ func (r *V1Repository) updateLocalSnapshot() (*v1manifest.Snapshot, error) {
 	return &snapshot, nil
 }
 
+func fnameWithVersion(fname string, version uint) string {
+	base := filepath.Base(fname)
+	dir := filepath.Dir(fname)
+
+	versionBase := strconv.Itoa(int(version)) + "." + base
+	return filepath.Join(dir, versionBase)
+}
+
 func (r *V1Repository) updateLocalRoot() error {
-	// When we save to disk need to save twice:
-	//filename := v1manifest.ManifestFilenameRoot
-	//err := local.SaveManifest(manifest, filename)
-	//version := manifest.Signed.Base().Version
-	//filename =  fmt.Sprintf("%v.%s", version, filename)
-	//err = local.SaveManifest(manifest, filename)
+	var root1 v1manifest.Root
+	_, err := r.local.LoadManifest(&root1)
+	if err != nil {
+		return err
+	}
+
+	var newRoots []v1manifest.Manifest
+
+	for version := root1.Version + 1; ; version++ {
+		var root2 v1manifest.Root
+		fname := fnameWithVersion(v1manifest.ManifestURLRoot, version)
+
+		reader, err := r.mirror.Fetch(fname, int64(maxRootSize))
+		if err != nil {
```

Comment:
I think we break just if the error is `NotFound`, and other errors should propagate to the caller.